code/lib/preview-api/template/stories/rendering.stories.ts

```diff
@@ -66,3 +66,31 @@ export const ChangeArgs = {
     await expect(button).toHaveFocus();
   },
 };
+
+let loadedLabel = 'Initial';
+
+/**
+ * This story demonstrates what happens when rendering (loaders) have side effects, and can possibly interleave with each other
+ * Triggering multiple force remounts quickly should only result in a single remount in the end
+ * and the label should be 'Loading... Done. Click Me' at the end. If loaders are interleaving it could result in a label of 'Loading... Done. Done. Done. Done. Click Me'
+ * Similarly, changing args rapidly should only cause one rerender at a time, producing the same result.
+ */
+export const SlowLoader = {
+  loaders: [
+    async () => {
+      loadedLabel = 'Loading...';
+      await new Promise((resolve) => setTimeout(resolve, 1000));
+      loadedLabel += ' Done.';
+      return { label: loadedLabel };
```

Comment:
Would it be more clear if this code was something like:

```ts
      loadedLabel = 'Loading...';
      await new Promise((resolve) => setTimeout(resolve, 1000));
      loadedLabel = loadedLabel === 'Loading...' ? 'Done' : 'Error';
      return { label: loadedLabel };
```

I'm fine if you prefer it this way.